src/audio/google/google_ctc_audio_processing_ipc3.c

```diff
@@ -39,6 +39,13 @@ int ctc_set_config(struct processing_module *mod, uint32_t param_id,
 			cd->reconfigure = true;
 		}
 		return 0;
+	case SOF_CTRL_CMD_SWITCH:
+		if (cdata->num_elems == 1) {
+			cd->enabled = cdata->chanv[0].value;
+			comp_info(dev, "ctc_set_config(), enabled = %d",
+				  cd->enabled);
+			return 0;
+		}
```

Comment:
is this intended - to fall through here? If so, please use `COMPILER_FALLTHROUGH`

Reply:
It was intended to fall through to `default` and returned as invalid control.
I changed to handle in a different error message to make it clear and more error-proof.

Thanks for pointing this out!

Reply:
right, this fixes it, thanks. Just a soft wish - in such cases SOF usually prefers no `else`. I.e.
```
if (x) {
    handle_x();
    return 0;
}
print("error");
return -Esuitableerror;
```
or perhaps
```
if (e) {
    print("error");
    return -Esuitableerror;
}
// proceed normally
```